graftm/pplacer.py

```diff
@@ -40,42 +40,103 @@ def alignment_merger(self, alignment_files, output_alignment_path):
                 for record in alignments: # For each record in the read list
                     record.id = record.id + '_' + str(file_number) # append the unique identifier to the record id
                 SeqIO.write(alignments, output, "fasta") # And write the reads to the file
-                alias_hash[str(file_number)] = {'output_path': os.path.join(os.path.dirname(alignment_file),'placements.jplace'),
+                alias_hash[str(file_number)] = {'output_path': os.path.join(os.path.dirname(alignment_file), 'placements.jplace'),
                                                 'place': []}
                 file_number += 1
         return alias_hash
+    
+    def convert_cluster_dict_keys_to_aliases(self, cluster_dict, alias_hash):
+        '''
+        Parameters
+        ----------
+        cluster_dict : dict
+            dictionary stores information on pre-placement clustering  
+        alias_hash : dict
+            Stores information on each input read file given to GraftM, the
+            corresponding reads found within each file, and their taxonomy
+    
+        Returns
+        --------
+        updated cluster_dict dict containing alias indexes for keys.
+        '''
+        output_dict = {}
+        directory_to_index_dict = {os.path.split(item["output_path"])[0] : key 
+                            for key, item in alias_hash.iteritems()}
+        for key, item in cluster_dict.iteritems():
+            cluster_file_directory = os.path.split(key)[0]
+            cluster_idx = directory_to_index_dict[cluster_file_directory]
+            output_dict[cluster_idx] = item
+        return output_dict
+            
+    
+    
+    def jplace_split(self, original_jplace, alias_hash, cluster_dict):
+        '''
+        To make GraftM more efficient, reads are dereplicated and merged into
+        one file prior to placement using pplacer. This function separates the
+        single jplace file produced by this process into the separate jplace 
+        files, one per input file (if multiple were provided) and backfills 
+        abundance (re-replicates?) into the placement file so analyses can be 
+        done using the placement files. 
+        
+        Parameters
+        ----------
+        original_jplace : dict (json)
+            json .jplace file from the pplacer step.
+        alias_hash : dict
+            Stores information on each input read file given to GraftM, the
+            corresponding reads found within each file, and their taxonomy
```

Comment:
I don't get this 'v3' thing. Using `n` instead of `nm` is still valid v3 format.

This section describes the input hash, which makes 'stores' unclear. Stored before input or that's what this function does? Give an example of before and after I reckon. Why does 'place' need to be given as an empty array?